src/lib/sort/sort-header.ts

```diff
@@ -0,0 +1,89 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {Component, Input, Optional} from '@angular/core';
+import {MdSort, MdSortable} from './sort';
+import {MdSortIntl} from './sort-intl';
+import {CdkColumnDef} from '../core/data-table/cell';
+import {SortDirection} from './sort-direction';
+import {coerceBooleanProperty} from '../core';
+import {getMdSortHeaderNotContainedWithinMdSortError} from './sort-errors';
+
+/**
+ * Applies sorting behavior (click to change sort) and styles to an element, including an
+ * arrow to display the current sort direction.
+ *
+ * Must be provided with an id and contained within a parent MdSort directive.
+ *
+ * If used on header cells in a CdkTable, it will automatically default its id from its containing
+ * column definition.
+ */
+@Component({
+  moduleId: module.id,
+  selector: '[md-sort-header], [mat-sort-header]',
+  templateUrl: 'sort-header.html',
+  styleUrls: ['sort-header.css'],
+  host: {
+    '(click)': '_sort.sort(this)',
+    '[class.mat-sort-header-sorted]': '_isSorted()',
+  }
+})
+export class MdSortHeader implements MdSortable {
+  /**
+   * ID of this sort header. If used within the context of a CdkColumnDef, this will default to
+   * the column's name.
+   */
+  @Input('md-sort-header') id: string;
+
+  /** Sets the position of the arrow that displays when sorted. */
+  @Input() arrowPosition: 'before' | 'after' = 'after';
+
+  /** Overrides the sort start value of the containing MdSort for this MdSortable. */
+  @Input('start') start: SortDirection;
+
+  /** Overrides the disable clear value of the containing MdSort for this MdSortable. */
+  @Input()
+  get disableClear() { return this._disableClear; }
+  set disableClear(v) { this._disableClear = coerceBooleanProperty(v); }
+  private _disableClear: boolean;
+
+  /** Overrides the reverse order value of the containing MdSort for this MdSortable. */
+  @Input()
+  get reverseOrder() { return this._reverseOrder; }
```

Comment:
I think having both of these is more confusing than helpful. The problem with `"reverse"` is that you have to know what the default is. Saying `start` or `startWith`, in my mind, better captures the developers intent